components/brave_vpn/brave_vpn_service.cc

```diff
@@ -900,6 +901,12 @@ void BraveVpnService::LoadPurchasedState() {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
   if (GetPurchasedStateSync() == PurchasedState::LOADING)
     return;
+  if (!IsNetworkAvailable()) {
+    VLOG(2) << __func__ << ": Network is not available, failed to connect";
+    SetPurchasedState(PurchasedState::NOT_PURCHASED);
+    UpdateAndNotifyConnectionStateChange(ConnectionState::CONNECT_FAILED);
```

Comment:
I think it would be better to notify this by monitoring network state instead of doing here. Is it possible?

Reply:
done